src/Avalonia.Controls/ColorPicker/ColorChangedEventArgs.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Portions of this source file are adapted from the WinUI project.
+// (https://github.com/microsoft/microsoft-ui-xaml)
+//
+// Licensed to The Avalonia Project under the MIT License.
+
+using System;
+using Avalonia.Media;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Holds the details of a ColorChanged event.
+    /// </summary>
+    /// <remarks>
+    /// HSV color information is intentionally not provided.
+    /// Use <see cref="Color.ToHsv()"/> to obtain it.
+    /// </remarks>
+    public class ColorChangedEventArgs : EventArgs
+    {
+        private Color _OldColor;
+        private Color _NewColor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ColorChangedEventArgs"/> class.
+        /// </summary>
+        public ColorChangedEventArgs()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ColorChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="oldColor">The old/original color from before the change event.</param>
+        /// <param name="newColor">The new/updated color that triggered the change event.</param>
+        public ColorChangedEventArgs(Color oldColor, Color newColor)
+        {
+            _OldColor = oldColor;
+            _NewColor = newColor;
+        }
+
+        /// <summary>
+        /// Gets the old/original color from before the change event.
+        /// </summary>
+        public Color OldColor
+        {
+            get => _OldColor;
+            internal set => _OldColor = value;
```

Comment:
Normally I would ask to use new ``` { get; init; }``` here (you could add **IsExternalInit** attribute polyfill for that).
But not today. Couple of days ago I tried to use **IsExternalInit** for new file picker API, and that broke iOS backend... So, yeah, not today.

Reply:
I left the code as-is from an earlier port here. I noticed the backing fields and like you had a thought to remove them but since they were already there didn't bother. Generally, I agree with you though that this can be simplified.

There is a little history here: In UWP the 'set' accessor is never exposed to C#; however, it is available and used in C++ (probably by setting the field directly). Therefore, in the direct port I had to add an `internal set` to avoid changing code elsewhere. The UWP code also doesn't have a constructor with parameters though, which is now added. So I think code can be reworked everywhere and remove the `internal set` and then the backing fields. I'll just use the new constructor instead.

Reply:
I revised the code using the constructor with parameters and removing the backing fields.